Userland/xargs.cpp

```
@@ -0,0 +1,186 @@
+            while (first_index.has_value()) {
+                auto current_index = first_index.value();
+                auto substring_until_replacement = arg.substring_view(previous_index, current_index);
+                parts.append(substring_until_replacement);
+                previous_index = current_index;
+                first_index = arg.substring_view(current_index + replace_view.length(), arg.length() - current_index - replace_view.length()).find_first_of(replace_view);
+            }
+            parts.append(arg.substring_view(previous_index + replace_view.length(), arg.length() - previous_index - replace_view.length()));
+        } else {
+            parts.append(arg);
+        }
+        argument_parts.append(move(parts));
+        ++index;
+    }
+
+    // Append a '{}' at the end if no arguments have replacements.
+    if (!some_argument_has_replacement) {
+        Vector<StringView> parts;
+        parts.append("");
+        parts.append("");
+        argument_parts.append(move(parts));
+    }
+
+    FILE* fp = stdin;
+    bool is_stdin = true;
+
+    if (StringView { "-" } != file_to_read) {
+        // A file was specified, try to open it.
+        fp = fopen(file_to_read, "r");
+        if (!fp) {
+            perror("fopen");
+            return 1;
+        }
+        is_stdin = false;
+    }
+
+    char* line = nullptr;
+    size_t buffer_size = 0;
+
+    StringBuilder builder;
+    Vector<char*> child_argv;
+
+    child_argv.resize(argument_parts.size() + 2);
+    child_argv[0] = const_cast<char*>(command);
+    child_argv[child_argv.size() - 1] = 0;
+
+    int devnull_fd = 0;
+
+    if (is_stdin) {
+        devnull_fd = open("/dev/null", O_RDONLY);
+        if (devnull_fd < 0) {
+            perror("open");
+            return 1;
+        }
+    }
+
+    for (;;) {
+        auto line_size = getdelim(&line, &buffer_size, entry_separator, fp);
+
+        if (line_size <= 0)
+            break;
```

This just swallows read errors doesn't it?

---

Ah, we should show the error too, yeah.